apps/backend/src/graphql/definitions/Team.ts

```diff
@@ -454,5 +559,169 @@ export const resolvers: IResolvers = {
       await deleteAccount({ id: args.input.accountId, user: ctx.auth?.user });
       return true;
     },
+    enableGitHubSSOOnTeam: async (_root, args, ctx) => {
+      if (!ctx.auth) {
+        throw unauthenticated();
+      }
+
+      const [installation, hasAccessToInstallation, teamAccount] =
+        await Promise.all([
+          GithubInstallation.query()
+            .findOne("githubId", args.input.ghInstallationId)
+            .where("deleted", false)
+            .throwIfNotFound(),
+          checkUserHasAccessToInstallation(
+            ctx.auth.user,
+            args.input.ghInstallationId,
+          ),
+          getWritableAccount({
+            id: args.input.teamAccountId,
+            user: ctx.auth.user,
+          }),
+        ]);
+
+      if (!hasAccessToInstallation) {
+        throw forbidden("User does not have access to GitHub installation");
+      }
+
+      const appOctokit = getAppOctokit();
+
+      const ghInstallation = await appOctokit.apps.getInstallation({
+        installation_id: installation.githubId,
+      });
+      const ghOrg = ghInstallation.data.account;
+      invariant(ghOrg, "GitHub Organization not found");
+      invariant("type" in ghOrg, "GitHub enterprise not supported");
+
+      const githubAccount = await getOrCreateGithubAccount(ghOrg);
+      invariant(githubAccount, "GitHub account not found");
+
+      const octokit = await getInstallationOctokit(installation.id);
+      invariant(octokit, "Invalid installation");
+
+      const { teamId } = teamAccount;
+      invariant(teamId, "Account teamId is undefined");
+
+      const newTeamUserInputs = await importOrgMembers({
+        octokit,
+        org: ghOrg.login,
+        teamId,
+        githubAccount,
+      });
+
+      const manager = teamAccount.$getSubscriptionManager();
+      const [plan, subscriptionStatus, subscription] = await Promise.all([
+        manager.getPlan(),
+        manager.getSubscriptionStatus(),
+        manager.getActiveSubscription(),
+      ]);
+
+      if (subscriptionStatus !== "active") {
+        throw forbidden("A valid subscription is required to enable SSO");
+      }
+
+      const priced = !plan?.githubSsoIncluded;
+
+      if (priced) {
+        if (!subscription) {
+          throw forbidden("A valid subscription is required to enable SSO");
+        }
+
+        if (!subscription.stripeSubscriptionId) {
+          throw forbidden("GitHub SSO is not available on your current plan");
```

Comment:
It's funny that, for the moment, GitHub SSO is not available for client that had subscribed via GitHub 🙃